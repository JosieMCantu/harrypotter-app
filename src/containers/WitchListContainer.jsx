import React, {useState, useEffect} from 'react';
import {WitchList} from '../components/presentational/WitchList';
import {useWitches} from '../hooks/thisWitch';

function WitchListContainer() {
    const {witches, loading} = useWitches();

    if(loading) { 
        return <h1>Loading...</h1>
    }

    return (
        <>
            <WitchList witches={witches}/>
        </>
    )
}
export default WitchListContainer;
