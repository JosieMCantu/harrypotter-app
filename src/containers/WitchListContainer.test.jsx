import React from 'react';
import {screen, render} from '@testing-library/react';
import WitchListContainer from '../containers/WitchListContainer';

describe('Harry Potter Witch Container', () => {
    it('shows a list of witches on the screen', async () => {
        render(<WitchListContainer/>)

        screen.getByText('Loading...')

        const ul = await screen.findByRole('list', {name: 'witches'})
        expect(ul).toMatchSnapshot();
    })
})
