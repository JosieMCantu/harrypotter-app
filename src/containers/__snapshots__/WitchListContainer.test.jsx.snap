// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Harry Potter Witch Container shows a list of witches on the screen 1`] = `
<ul
  aria-label="witches"
>
  <li>
    <img
      alt="Harry Potter"
      src="http://hp-api.herokuapp.com/images/harry.jpg"
    />
    <p>
      Name: 
      Harry Potter
    </p>
    <p>
      Patronus: 
      stag
    </p>
  </li>
  <li>
    <img
      alt="Hermione Granger"
      src="http://hp-api.herokuapp.com/images/hermione.jpeg"
    />
    <p>
      Name: 
      Hermione Granger
    </p>
    <p>
      Patronus: 
      otter
    </p>
  </li>
  <li>
    <img
      alt="Ron Weasley"
      src="http://hp-api.herokuapp.com/images/ron.jpg"
    />
    <p>
      Name: 
      Ron Weasley
    </p>
    <p>
      Patronus: 
      Jack Russell terrier
    </p>
  </li>
  <li>
    <img
      alt="Draco Malfoy"
      src="http://hp-api.herokuapp.com/images/draco.jpg"
    />
    <p>
      Name: 
      Draco Malfoy
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Minerva McGonagall"
      src="http://hp-api.herokuapp.com/images/mcgonagall.jpg"
    />
    <p>
      Name: 
      Minerva McGonagall
    </p>
    <p>
      Patronus: 
      tabby cat
    </p>
  </li>
  <li>
    <img
      alt="Cedric Diggory"
      src="http://hp-api.herokuapp.com/images/cedric.png"
    />
    <p>
      Name: 
      Cedric Diggory
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Cho Chang"
      src="http://hp-api.herokuapp.com/images/cho.jpg"
    />
    <p>
      Name: 
      Cho Chang
    </p>
    <p>
      Patronus: 
      swan
    </p>
  </li>
  <li>
    <img
      alt="Severus Snape"
      src="http://hp-api.herokuapp.com/images/snape.jpg"
    />
    <p>
      Name: 
      Severus Snape
    </p>
    <p>
      Patronus: 
      doe
    </p>
  </li>
  <li>
    <img
      alt="Rubeus Hagrid"
      src="http://hp-api.herokuapp.com/images/hagrid.png"
    />
    <p>
      Name: 
      Rubeus Hagrid
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Neville Longbottom"
      src="http://hp-api.herokuapp.com/images/neville.jpg"
    />
    <p>
      Name: 
      Neville Longbottom
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Luna Lovegood"
      src="http://hp-api.herokuapp.com/images/luna.jpg"
    />
    <p>
      Name: 
      Luna Lovegood
    </p>
    <p>
      Patronus: 
      hare
    </p>
  </li>
  <li>
    <img
      alt="Ginny Weasley"
      src="http://hp-api.herokuapp.com/images/ginny.jpg"
    />
    <p>
      Name: 
      Ginny Weasley
    </p>
    <p>
      Patronus: 
      horse
    </p>
  </li>
  <li>
    <img
      alt="Sirius Black"
      src="http://hp-api.herokuapp.com/images/sirius.JPG"
    />
    <p>
      Name: 
      Sirius Black
    </p>
    <p>
      Patronus: 
      hare
    </p>
  </li>
  <li>
    <img
      alt="Remus Lupin"
      src="http://hp-api.herokuapp.com/images/lupin.jpg"
    />
    <p>
      Name: 
      Remus Lupin
    </p>
    <p>
      Patronus: 
      wolf
    </p>
  </li>
  <li>
    <img
      alt="Arthur Weasley"
      src="http://hp-api.herokuapp.com/images/arthur.jpg"
    />
    <p>
      Name: 
      Arthur Weasley
    </p>
    <p>
      Patronus: 
      weasel
    </p>
  </li>
  <li>
    <img
      alt="Bellatrix Lestrange"
      src="http://hp-api.herokuapp.com/images/bellatrix.jpg"
    />
    <p>
      Name: 
      Bellatrix Lestrange
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Lord Voldemort"
      src="http://hp-api.herokuapp.com/images/voldemort.jpg"
    />
    <p>
      Name: 
      Lord Voldemort
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Horace Slughorn"
      src="http://hp-api.herokuapp.com/images/slughorn.JPG"
    />
    <p>
      Name: 
      Horace Slughorn
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Kingsley Shacklebolt"
      src="http://hp-api.herokuapp.com/images/kingsley.jpg"
    />
    <p>
      Name: 
      Kingsley Shacklebolt
    </p>
    <p>
      Patronus: 
      lynx
    </p>
  </li>
  <li>
    <img
      alt="Dolores Umbridge"
      src="http://hp-api.herokuapp.com/images/umbridge.jpg"
    />
    <p>
      Name: 
      Dolores Umbridge
    </p>
    <p>
      Patronus: 
      persian cat
    </p>
  </li>
  <li>
    <img
      alt="Lucius Malfoy"
      src="http://hp-api.herokuapp.com/images/lucius.jpg"
    />
    <p>
      Name: 
      Lucius Malfoy
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Vincent Crabbe"
      src="http://hp-api.herokuapp.com/images/crabbe.jpg"
    />
    <p>
      Name: 
      Vincent Crabbe
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Gregory Goyle"
      src="http://hp-api.herokuapp.com/images/goyle.jpg"
    />
    <p>
      Name: 
      Gregory Goyle
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Mrs Norris"
      src="http://hp-api.herokuapp.com/images/norris.JPG"
    />
    <p>
      Name: 
      Mrs Norris
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
  <li>
    <img
      alt="Argus Filch"
      src="http://hp-api.herokuapp.com/images/filch.jpg"
    />
    <p>
      Name: 
      Argus Filch
    </p>
    <p>
      Patronus: 
      
    </p>
  </li>
</ul>
`;
